### DISCUSSION TITLE
Android Automatic OTP Retrival

### DISCUSSION THREAD

**Discussion by premdasvm** [invertase/react-native-firebase#4869] [Q&A]
hey guys, How can I auto fetch the OTP from Firebase Phone Authentication.? How to use Phone Auth with the SMS Retriever API .?

**Comment by mikehardy** [comment]
I believe it does it by default on Android. It's the source of some confusion in fact as the OTP is consumed but if you don't subscribe to the user-changed auth listener you won't know it. It's all in the docs.

How did it go when you tried it on a real android device?

**Reply by atharva-bhange** [reply]
Sorry but I have been having problem with this a lot. Just to clarify to get function triggering on otp sms received i have to subscribe to https://rnfirebase.io/reference/auth#onUserChanged this right according to your comment? 

In my case it didn't trigger when message was received 
Secondly, when it did trigger it used to log
```json
{
   "displayName":null,
   "email":null,
   "emailVerified":false,
   "isAnonymous":false,
   "metadata":{
      "creationTime":1629018723310,
      "lastSignInTime":1629018723310
   },
   "phoneNumber":"+919623550815",
   "photoURL":null,
   "providerData":[
      [
         "Object"
      ]
   ],
   "providerId":"firebase",
   "tenantId":null,
   "uid":"MDFsZtK8nzf60tnHz7Tb6IajIPp1"
}
```
So if possible can you please guide me towards how to subscribe to event listener which will trigger on firebase sms received. Or direct me to proper documentation.

I was trying to find something like this https://firebase.google.com/docs/auth/android/phone-auth#onverificationcompletedphoneauthcredential how they have shown for android for react native ?

I am using 
```json
{
    "@react-native-firebase/app": "^11.2.0",
    "@react-native-firebase/auth": "^11.2.0",
}
```


Thank you in advance for the help!

**Comment by musabgulfam** [comment]
In my case, how do I disable this functionality?

**Reply by mikehardy** [reply]
It's build in to the firebase-android-sdk library, I'm not sure it will be easy. Check upstream in their repo, and I think it might be finding the specific android permission that allows it, and in AndroidManifest.xml using a "tools:remove" line on the permission to edit it out of the combined manifest during android app build. This is vague though: you must go to the upstream repository and investigate, then check it. Using Android Studio to look at the combined AndroidManifest.xml file may give you a clue as to what component / permission is doing it so you can modify the behavior

**Comment by parthya** [comment]
Is there anyway to know if it working in background, so that we can display a message for user else display manual input.

Because it works on some phones and doesn't on older phones.
